### ISSUE TITLE
Cant get search bar to show in top nav

### ISSUE THREAD

**Issue by dabeeeenster** [facebook/docusaurus#4975]
### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.

### Description
I'm trying to get Algolia Docsearch integrated following [these docs](https://docusaurus.io/docs/search). I've added 

```
algolia: {            
  apiKey: '<Api Key>',           
  indexName: 'flagsmith',        
}
```

To my config but nothing is showing up in the nav bar. There is no info in the documentation around how to get the search widget to show up in the top nav either. 

## Expected behavior

I was expecting to either see the search bar show up after adding the Docsearch config, or I would have had to enable a component somewhere but this is not mentioned in the docs at all? 

## Actual Behavior

Nothing shows up

## Your Environment

- Public source code: https://github.com/Flagsmith/flagsmith-docs
- Public site URL: https://docs.flagsmith.com/
- Docusaurus version used:2.0.0-beta.0

**Comment by dabeeeenster** [comment]
OK I fixed this with `npm install --save @docusaurus/theme-search-algolia` - this should be added to the docs maybe?

**Comment by slorber** [comment]
@dabeeeenster 

The preset does: `themeConfig.algolia && require.resolve('@docusaurus/theme-search-algolia'),`

If you restart your site with the algolia config you should get an error on startup:

```
Error: Cannot find module '@docusaurus/theme-search-algolia'
```

Will mention it in the doc here: https://github.com/facebook/docusaurus/pull/4976

**Comment by dabeeeenster** [comment]
👍

**Comment by altafakhai** [comment]
> OK I fixed this with `npm install --save @docusaurus/theme-search-algolia` - this should be added to the docs maybe?

Hey Ben, I am facing a similar issue and not able to resolve this by running the above command! I had created a Docusaurus site a month back and have got the API Key and Index from Algolia. Didn't get the Search bar similar to what you had faced. Tried running the above command, but still no search available on the top-right. 

Request you to please help me out with it and let me know what you did to resolve the issue!

Regards,
Altaf

**Comment by dabeeeenster** [comment]
Yeah I've no idea - sorry! It just started showing up. I'm not sure what is triggering this but the docs could be improved around this feature.

**Comment by altafakhai** [comment]
Thanks for your response, Ben! 

Yes, the documentation needs some detailing about the Search bar, its integration, and troubleshooting. Will keep scouting for answers! 

Your site looks great. Amazing work!

**Comment by slorber** [comment]
Hi

Not 100% sure but it might be related to a caching issue we have in Webpack 5, preventing the usage of newly added theme components (and swizzled components).

Could be fixed in beta.2 by https://github.com/facebook/docusaurus/pull/5047, so let's see if people still have this problem after we release it

**Comment by altafakhai** [comment]
Hey, @slorber! Thank you so much for your help!

The workaround did fix up the issue! However, it didn't work with one of my previous samples which I had created sometime back. I guess I ran multiple commands to fix it. Just wanted to check if the sequence is right to create a new Docusaurus site:

1) Create a new Docusaurus site using - `npx @docusaurus/init@latest init my-website classic`
2) Then add the Search feature - `npm install --save @docusaurus/theme-search-algolia`
3) Then one might need to clear up the cache, so use `yarn docusaurus clear`
4) Add the Algolia API Key and Index Name, and make other modifications as required.

Let me know if the sequence is correct or I am doing too much here.

Looking forward to your response.

Altaf
